pkg/arvo/app/metadata-store.hoon

```diff
@@ -0,0 +1,172 @@
+::  metadata-store: data store for application metadata and mappings 
+::  between groups and application data (chatrooms, publish notebooks, etc)
+::
+/-  *metadata-store
+/+  default-agent
+|%
++$  card  card:agent:gall
+::
++$  versioned-state
+  $%  state-zero
+  ==
+::
++$  state-zero
+  $:  %0
+      =associated
+      group-indices=(jug group-path [app-name app-path])
+      app-name-indices=(jug app-name [group-path app-path])
+      app-path-indices=(map [app-name app-path] group-path)
```

Comment:
Are you saying that app-path-indices should be a jug? Until we have user stories around multiple groups associated with a singular app-path, I beg to differ.

Reply:
That's what I'm saying.

The `associations` data structure has keys of `[group-path app-name app-path]`. This means that you can issue `%create` actions to store, say, both `[/a %x /y]` and `[/b %x /y]`. Since these both point to the resource `[%x /y]`, the latter action will _overwrite_ the former one in `app-path-indices`, even though they're both getting stored in `associations`. Your indices are no longer accurate!

I understand we have no use cases for "multiple groups per resource" right now, but if we don't let you get into that situation in the first place I see no harm in having the store just support it already, so we don't have to go back and change this in the future.

If we insist on not supporting it on the metadata-store level, then we should add a check for `%create` actions that prevents such "duplicates". (The questions for apps that _rely_ on this store for state storage then becomes, what if it doesn't want to store my state? But if they shouldn't be getting into that position in the first place, maybe that's fine...)

Reply:
Okay, changed it to a jug.